src/OmniSharp.Cake/CachingScriptMetadataResolver.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Scripting;
+
+namespace OmniSharp.Cake
+{
+    internal class CachingScriptMetadataResolver : MetadataReferenceResolver
+    {
+        private static readonly Dictionary<string, ImmutableArray<PortableExecutableReference>> DirectReferenceCache = new Dictionary<string, ImmutableArray<PortableExecutableReference>>();
+        private static readonly Dictionary<string, PortableExecutableReference> MissingReferenceCache = new Dictionary<string, PortableExecutableReference>();
```

Comment:
Should either of these caches be case-insensitive on the key?

Reply:
Dunno, this was copy-pasted-with-pride from what used to be in OmniSharp.Script. Let’s ask @filipw?